File: src/transactions/test/wasm/test_invoke_contract.zig
extern fn log_value(i64) i64;
extern fn map_new() i64;
extern fn map_put(i64,i64,i64) i64;
extern fn map_get(i64,i64) i64;
extern fn get_current_ledger_num() i64;

export fn invoke(arg: i64) i64 {
    _ = log_value(arg);
    _ = log_value(get_current_ledger_num());
    return arg + 1;
}

File: src/transactions/test/wasm-rust/test_invoke_contract.rs
#![allow(dead_code)]

extern {
    fn log_value(v:i64) -> i64;
    fn map_new() -> i64;
    fn map_put(m:i64,k:i64,v:i64) -> i64;
    fn map_get(m:i64,k:i64) -> i64;
    fn get_current_ledger_num() -> i64;
}

#[no_mangle]
pub unsafe fn invoke(arg: i64) -> i64 {
    let _ = log_value(arg);
    let _ = log_value(get_current_ledger_num());
    return arg + 1;
}
